src/aws/flb_aws_util.c

```diff
@@ -863,7 +863,9 @@ flb_sds_t flb_get_s3_key(const char *format, time_t time, const char *tag,
         sprintf(seq_index_str, "%"PRIu64, seq_index);
         seq_index_str[seq_index_len] = '\0';
         tmp_key = replace_uri_tokens(s3_key, INDEX_STRING, seq_index_str);
-
+        if (!tmp_key) {
```

Comment:
I think you're missing a `flb_free(seq_index_str);` there.
Other than that, would you mind change that comparison to `if (tmp_key == NULL) {` instead? I'd really appreciate it.

Reply:
Am happy to change the formating, but this pattern is consistently used in `flb_aws_util.c`:

https://github.com/fluent/fluent-bit/blob/7f8f1600cbd2cace51a8eb0184f3cedc26d0339d/src/aws/flb_aws_util.c#L843-L845

https://github.com/fluent/fluent-bit/blob/7f8f1600cbd2cace51a8eb0184f3cedc26d0339d/src/aws/flb_aws_util.c#L885-L887

Do you still want me to change the pattern?

Reply:
Huh, hard question. I think I would, eventually I'd like all the code to be more explicit / clear and even though this would "break" the flow I think it's worth it.